packages/vm/lib/state/stateManager.ts

```diff
@@ -499,7 +488,11 @@ export default class DefaultStateManager implements StateManager {
    */
   async accountIsEmpty(address: Buffer): Promise<boolean> {
     const account = await this.getAccount(address)
-    return account.isEmpty()
+    // Note: ethereumjs-util account.isEmpty() additionally checks for empty stateRoot which differs here.
+    // EIP-161: "An account is considered empty when it has no code and zero nonce and zero balance."
+    return (
+      account.nonce.isZero() && account.balance.isZero() && account.codeHash.equals(KECCAK256_NULL)
+    )
```

Comment:
Ok, we should I would assume remove the `stateRoot` check again? 🤔

Reply:
yes I am still discussing with @jochem-brouwer but I think that is what we will end up doing. this is still safe to merge in, and once updated in util then I can update here back to `account.isEmpty()`